coding-test/12일차/calculator.js

```diff
@@ -0,0 +1,85 @@
+// 로직구성
+//0. 텍스트를 입력하고, 입력을 누르면 자동으로 값이 계산되게 하는 프로그램 작성
+//0. 기능별로 함수를 나눠서 최대한 작게 만들기
+//0. 함수의 길이가 20이 넘지 않도록 한다.
+//0. 들여쓰기 4가 안 넘도록 구성
+
+//1. 입력을 누르면 값이 튀어나오게 함수구성
+//2. exit라는 값을 적고 입력을 눌렀을 때, 종료 경고창나오게
+//3. +-*/ 계산별로 함수 구현 -> 배열에 넣고 계산
+
+//텍스트 입력 시 출력되는 함수
+function print_text() {
+    let text_value = document.getElementById('text').value;
+    input_exit(text_value);
+    let total = calculator(text_value);
+    document.getElementById('result').innerHTML = total; //입력 그대로 출력
+}
+
+//exit 입력 시 종료
+function input_exit(text_value) {
+    if(text_value === 'exit') {
+        alert('계산기 종료');
+    }
+}
+
+//받은 text_value를 배열 형태롤 저장하는 함수
+function text_makeArr(text_value) {
+    const text_arr = text_value.match(/[\d\+\*\/\-]/g);
+    return text_arr;
+}
+
+//계산기
+function calculator(text_value) {
+    const text_arr = text_makeArr(text_value);
+    for(let i = 0; i < text_arr.length; i++) {
+        mul_div(text_arr); //우선순위1
+        add_sub(text_arr); //우선순위2
+    }
+    // console.log(Number(text_arr[0].toFixed(2)));
+    return Number(text_arr[0].toFixed(2))
+}
+
+
+//계산처리 후 배열에 추가/삭제
+function cal(text_arr, i, temp) {   
+    console.log(text_arr[i-1]);
+    text_arr[i-1] = temp;   //연산된 값을 추가
+    console.log(text_arr[i]);
+    console.log(text_arr[i+1]);
+    // console.log(text_arr.indexOf(text_arr[i-1])); //인덱스 확인
+    text_arr.splice(text_arr.indexOf(text_arr[i]),2);   //연산 후 값 삭제
+    console.log(text_arr);
+    return text_arr;
+}
```

Comment:
```suggestion
function cal(textList, index, temp) {   
   const result = [...textList];
   result[i - 1] = temp;
   result.splice(result.indexOf(result[i], 2)); 
    return result;
}
```

인자로 받은 값을 바로 수정하는것은 좋지 못해요.
인자로 배열이나 객체를 받았을때, 그것을 바로 사용하는것이 아니라 함수 안에서 복사하여 새로 생성 후에 이후 로직을 작성해주시는게 더 좋습니다.
순수함수의 개념에 대해서 한번 찾아보시면 도움이 될거에요.

Reply:
좋은 조언 감사합니다 열심히 해보겠습니다!